modules/datastore/tests/src/Unit/DataDictionary/AlterTableQuery/BuilderTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Drupal\Tests\datastore\Unit\DataDictionary\AlterTableQuery;
+
+use Drupal\Core\Database\Connection;
+use Drupal\common\Storage\DatabaseConnectionFactoryInterface;
+use Drupal\datastore\DataDictionary\AlterTableQueryBuilderInterface;
+use Drupal\datastore\DataDictionary\AlterTableQueryInterface;
+use Drupal\datastore\DataDictionary\AlterTableQuery\Builder;
+use Drupal\datastore\DataDictionary\AlterTableQuery\MySQLQuery;
+
+use MockChain\Chain;
+use PDLT\ConverterInterface;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Unit tests for Drupal\datastore\DataDictionary\AlterTableQuery\Builder.
+ */
+class BuilderTest extends TestCase {
+
+  /**
+   * Test Builder.
+   */
+  public function test() {
+
+    $connection = (new Chain($this))
+      ->add(DatabaseConnectionFactoryInterface::class, 'setConnectionTimeout', DatabaseConnectionFactoryInterface::class)
+      ->addd('getConnection', Connection::class)
+      ->getMock();
+    $converter = (new Chain($this))
+      ->add(ConverterInterface::class)
+      ->getMock();
+
+    $builder = new Builder($connection, $converter, MySQLQuery::class);
+
+    // Test Builder's setConnectionTimeout() returns what's expected.
+    $result = $builder->setConnectionTimeout(1);
+    $this->assertTrue(is_a($result, AlterTableQueryBuilderInterface::class));
```

Comment:
Should these assertions check that the QueryBuilder did anything? It doesn't really seem to be testing anything other than the return type

Reply:
It's checking that an AlterTableQuery object was built, but no properties or methods are public, except for `->execute()`, which is being tested in `MySQLQueryTest`.

We could potentially create a AlterTableQuery mock object which exposes these protected properties, but at that point were only adding additional test coverage to the constructor, which is already being tested in `MySQLQueryTest`.